test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -2131,5 +2132,97 @@ def test_clbit_conditional(self):
         )
 
 
+class TestSwapNodes(QiskitTestCase):
+    """Test Swapping connected nodes."""
+
+    def test_1q_swap_fully_connected(self):
+        """Test swapping single qubit gates"""
+        from qiskit.converters import dag_to_circuit
+
+        qreg = QuantumRegister(1)
+        circuit = QuantumCircuit(qreg)
+        circuit.rx(pi, qreg[0])
+        circuit.rx(pi / 2, qreg[0])
+        dag = circuit_to_dag(circuit)
+        op_nodes = dag.op_nodes()
```

Comment:
This dance might be a little easier if we just built the `DAGCircuit` directly - the return value from `apply_operation_back` is the node index we're inferring here.